src/test/Test.FAKECore/FSIHelperSpecs.cs

```diff
@@ -0,0 +1,230 @@
+                }
+            };
+
+        It should_load_file =
+            () =>
+            {
+                var mainPath = Path.GetTempFileName() + ".fsx";
+                var loadedPath = Path.GetTempFileName() + ".fsx";
+                try
+                {
+                    var mainScript = "printf \"main\"\n#load \"" + loadedPath.ToString().Replace("\\", "/") + "\"";
+                    var loadedScript = "printf \"loaded;\"";
+                    File.WriteAllText(mainPath, mainScript);
+                    File.WriteAllText(loadedPath, loadedScript);
+
+                    RunExplicit(mainPath, "", false).ShouldEqual("loaded;main");
+                }
+                finally
+                {
+                    File.Delete(mainPath);
+                    File.Delete(loadedPath);
+                }
+
+            };
+
+        It should_change_hash_when_loaded_file_changes =
+            () =>
+            {
+                var middleDirName = Guid.NewGuid().ToString();
+                var middleDirPath = Path.Combine(Path.GetTempPath(), middleDirName);
+                Directory.CreateDirectory(middleDirPath);
+
+                var mainPath = (Path.GetTempFileName() + ".fsx");
+                var middle1Path = Path.Combine(middleDirPath, "middle1.fsx").Replace("\\", "/");
+                var middle2Path = Path.Combine(middleDirPath, "middle2.fsx").Replace("\\", "/");
+                var lastPath = (Path.GetTempFileName() + ".fsx").Replace("\\", "/");
+
+                var lastScript = "printfn \"foobar\"";
+                var middle2Script = "#load @\"" + lastPath + "\"";
+                var middle1Script = "#load \"\"\"middle2.fsx\"\"\"";
+                var mainScript = "#load \"" + middleDirName + "/middle1.fsx\"";
+
+                File.WriteAllText(mainPath, mainScript);
+                File.WriteAllText(middle1Path, middle1Script);
+                File.WriteAllText(middle2Path, middle2Script);
+                File.WriteAllText(lastPath, lastScript);
+
+                var scriptContents = FSIHelper.getAllScripts(mainPath);
+                var hash = FSIHelper.getScriptHash(scriptContents);
+
+                File.WriteAllText(lastPath, "printfn \"foobarbaz\"");
+                
+                scriptContents = FSIHelper.getAllScripts(mainPath);
+                var newHash = FSIHelper.getScriptHash(scriptContents);
+                hash.ShouldNotEqual(newHash);
+            };
+        It should_get_included_assemblies =
+            () =>
+            {
+                var script = 
+                    "#r \"justname\"\n" + 
+                    "#r \"./relative/path\"\n" +
+                    "#r \"C:/absolute/path\"";
+
+                var included = FSIHelper.getIncludedAssembly(script);
+
+                included.ShouldEqual(new string[] { "justname", "./relative/path", "C:/absolute/path" });
+            };
+        //It should_load_assembly =
```

Comment:
Ok so could you please remove these?
On Jul 13, 2015 18:36, "Xavier Zwirtz" <notifications@github.com> wrote:

> In src/test/Test.FAKECore/FSIHelperSpecs.cs
> <https://github.com/fsharp/FAKE/pull/859#discussion_r34482490>:
>
> > +                var newHash = FSIHelper.getScriptHash(scriptContents);
> > +                hash.ShouldNotEqual(newHash);
> > +            };
> > +        It should_get_included_assemblies =
> > +            () =>
> > +            {
> > +                var script =
> > +                    "#r \"justname\"\n" +
> > +                    "#r \"./relative/path\"\n" +
> > +                    "#r \"C:/absolute/path\"";
> > +
> > +                var included = FSIHelper.getIncludedAssembly(script);
> > +
> > +                included.ShouldEqual(new string[] { "justname", "./relative/path", "C:/absolute/path" });
> > +            };
> > +        //It should_load_assembly =
>
> I was originally parsing out the included assemblies in the script files,
> that was the test for it. I switched to saving a list of all loaded
> assemblies paths and loading those on next run. Made the test not needed.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/fsharp/FAKE/pull/859/files#r34482490>.
>